### PR TITLE
docs($root events): updated documentation

### LINKED ISSUE #2253: Is it possible somehow to listen events globally? For example, when popover is shown/hidden
For example there is a popover with `ref="myPopover"`.
Is there any way to listen `myPopover` for such events as `shown`/`hidden` globally, something like `addEventListener` for `shown` and `hidden` methods?

From the source code seems that it is possible, but I do not understand how to connect to the internal event system in order to listen it.

I need to know if `myPopover` is shown/hidden far far away from a component where `myPopover` lives and I do not want to use `vuex` + `@shown="callSomeFunction"`.

Thank.

### PR DESCRIPTION
**Pull request by TitanFighter**
## Description of Pull Request:

1. Adds info "How to use $root events" for the next components:
- collapse;
- modal;
- popover;
- table;
- tooltip.

2. Creates a separate section inside appropriate `Component Reference` in order to group $root event emitters, instead of searching them all over the document.

## PR checklist:

**What kind of change does this PR introduce?** (check at least one)
- [ ] Bugfix
- [ ] Feature
- [ ] Enhancement to an existing feature
- [ ] ARIA accessibility
- [x] Documentation update
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)
- [ ] Yes
- [x] No

If yes, please describe the impact:

**The PR fulfills these requirements:**
- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e. fixes #2253 )
- [x] The PR should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] PR titles should following the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. "fix(alert): not alerting during SSR render", "docs(badge): Updated pill examples, fix typos", "chore: fix typo in docs", etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**
- [x] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] New/updated tests are included and passing (if required)
- [x] Existing test suites are passing
- [x] The changes have not impacted the functionality of other components or directives 
- [ ] ARIA Accessibility has been taken into consideration (does it affect screen reader users or keyboard only users? clickable items should be in the tab index, etc)

**If adding a new feature, or changing the functionality of an existing feature, the PR's description above includes:**
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

**Comment by TitanFighter** [comment]
@tmorehouse
I have a problem with modals. I expected that `bv::modal::show` should work the same way as `bv::popover::show` or `bv::tooltip::show` (ie should provide bvObject), but when I open a modal `this.$root.$on("bv::modal::show", function(...params) {}` is simply fired without providing any params, so basically there is no way to know anything about the modal which has been opened.

[Demo](https://codesandbox.io/s/38q3zozzp5).

For this reason in the modal doc I added `bv::show::modal` and `bv::hide::modal` twice for `$root.$emit` and for `$root.$on`, because in the second case it is possible to get the modal id which is opened. It seems that implementation of popover + tooltip differ from modal.

**Comment by TitanFighter** [comment]
@tmorehouse
I have a problem with modals. I expected that `bv::modal::show` should work the same way as `bv::popover::show` or `bv::tooltip::show` (ie should provide bvObject), but when I open a modal `this.$root.$on("bv::modal::show", function(...params) {}` is simply fired without providing any params, so basically there is no way to know anything about the modal which has been opened.

[Demo](https://codesandbox.io/s/38q3zozzp5).

For this reason in the modal doc I added `bv::show::modal` and `bv::hide::modal` twice for `$root.$emit` and for `$root.$on`, because in the second case it is possible to get the modal id which is opened. It seems that implementation of popover + tooltip differ from modal.

File: src/components/modal/README.md [hunk 1]
```diff
@@ -471,6 +472,35 @@ To disable stacking for a specific modal, just set the prop `no-stacking` on the
 - The opaque backdrop will appear progressively darker for each modal that is opened. This is expected behaviour as each backdrop is opened over top the other backdrops.
 - For multiple modals to stack properly, they **must** be defined in the document in the order they will be opened, otherwise a newly opened modal may appear hidden or obscured by a currently open modal.
 
+
+## Listen for a modal changes via $root events
```
Comment (tmorehouse) [inline src/components/modal/README.md]: This should be noted that it is the `v-b-toggle` directive that emits this event, not he modal itself.

If a modal is opened programatically via $refs methods (i.e. `this.$refs.mymodal.show()`), or using the v-model, etc, this event **will not** be emitted.

File: src/components/modal/README.md [hunk 2]
```diff
@@ -471,6 +472,35 @@ To disable stacking for a specific modal, just set the prop `no-stacking` on the
...
+  this.$root.$on("bv::show::modal", function(modalId, elIDtoFocusOnClose) { {
+    console.log("modalId:", modalId);
+    console.log("elIDtoFocusOnClose:", elIDtoFocusOnClose);
+  });
+}
+```
+
+where `modalId` is modal id to be shown; `elIDtoFocusOnClose` is id of element to return focus to once the modal is closed.
+
+To listen for hiding any modals, use:
```
Comment (tmorehouse) [inline src/components/modal/README.md]: Again, as above this is misleading as it is **not** the modal that emits this event, but rather `v-b-toggle` (or user code that wants to programmatically hode a modal).

As above, this is a control event, not a status event!

File: src/components/modal/package.json [hunk 3]
```diff
@@ -77,6 +77,56 @@
                         "description": "BvEvent object. Call bvEvt.preventDefault() to cancel hide"
                     }
                 ]
+            },
+            {
+                "event": "bv::show::modal",
+                "description": "Emitted on $root when modal is about to be shown.",
```
Comment (tmorehouse) [inline src/components/modal/package.json]: Incorrect.  this is not emitted by modal at all.  But is used to control the modal, and should not be listed under this section.

File: src/components/collapse/package.json [hunk 4]
```diff
@@ -10,27 +10,43 @@
...
+                "description": "Emitted when collapse has started to close"
             },
             {
                 "event": "hidden",
-                "description": "emitted when collaspe has finished closing"
+                "description": "Emitted when collapse has finished closing"
             },
+            {
+                "event": "bv::collapse::state",
+                "description": "Emitted when collapse has changed its state",
```
Comment (tmorehouse) [inline src/components/collapse/package.json]: This description should note that the event is emitted on `$root` not the component instance.

File: src/components/modal/README.md [hunk 5]
```diff
@@ -471,6 +472,35 @@ To disable stacking for a specific modal, just set the prop `no-stacking` on the
 - The opaque backdrop will appear progressively darker for each modal that is opened. This is expected behaviour as each backdrop is opened over top the other backdrops.
 - For multiple modals to stack properly, they **must** be defined in the document in the order they will be opened, otherwise a newly opened modal may appear hidden or obscured by a currently open modal.
 
+
+## Listen for a modal changes via $root events
+
+To listen for opening any modals, use:
```
[the remark this answers is not in the document]
Comment (tmorehouse) [reply src/components/modal/README.md]: The BvEvent object passed with `bv:modal:show` (etc) includes information about which modal is opening/hiding.

[the remark this answers is not in the document]
Comment (tmorehouse) [reply src/components/modal/README.md]: `bv::show::modal` and `bv::hide::modal` _should_ be in the docs, in the area around v-model and methods (`hide()`/`show()`). I'll have to check that.

[the remark this answers is not in the document]
Comment (TitanFighter) [reply src/components/modal/README.md]: `bv::show::modal` and `bv::hide::modal` exist, but not `bv::modal::show`, `bv::modal::shown`, `bv::modal::hide`, `bv::modal::hidden`.

**Comment by TitanFighter** [comment]
I did not add `bvEvent` to the docs because, as you can find in the [Demo](https://codesandbox.io/s/38q3zozzp5) linked above, `bv::modal::show/shown/hide/hidden` does not receive it (as expected).

**Comment by tmorehouse** [comment]
See the following lines of code: https://github.com/bootstrap-vue/bootstrap-vue/blob/dev/src/components/modal/modal.js#L545-L549

```js
    emitEvent (bvEvt) {
      const type = bvEvt.type
      this.$emit(type, bvEvt)
      this.$root.$emit(`bv::modal::${type}`, bvEvt)
    },
```

The `bvEvt` object is passed to the `$root.$emit(...)`

**Comment by TitanFighter** [comment]
Just checked locally... Everything is ok. Maybe a glitch in CodeSandbox.

**Comment by tmorehouse** [comment]
I'm seeing a bunch of errors in timescript on codesandbox

**Pull request merged**